Hush/Assets/Scripts/Weapon/PlayerWeaponController.cs

```diff
@@ -18,52 +18,54 @@ void Start()
     {
         equippedWeapon = sword;
         characterStats = new CharacterStats(10, 10);
-        
+
         // add any bonuses related to specific weapons
-        characterStats.AddBonus( StatType.Strength , sword.BonusDamage);
-        characterStats.AddBonus( StatType.SpellPower , spell.BonusDamage);
+        characterStats.AddBonus(StatType.Strength, sword.BonusDamage);
+        characterStats.AddBonus(StatType.SpellPower, spell.BonusDamage);
     }
 
     void OnEnable()
     {
         // TODO update to use enum after merge
-        input.reference.actions[Actions.LightAttack].performed += PerformWeaponAttack; 
+        input.reference.actions[Actions.LightAttack].performed += PerformWeaponAttack;
         input.reference.actions[Actions.HeavyAttack].performed += PerformWeaponSpecialAttack;
         input.reference.actions[Actions.SwitchWeapon].performed += SwitchWeapon;
     }
 
     private void SwitchWeapon(InputAction.CallbackContext callbackContext)
     {
-        equippedWeapon = (equippedWeapon == sword) ? (IWeapon) spell : sword;
+        equippedWeapon = (equippedWeapon.Name == sword.name) ? (IWeapon) spell : sword;
```

Comment:
idk if it's just because it bugs me or if it's a valid thing to denote, but why is the "name" property of both equippedWeapon and sword not capialized the same way?

Reply:
ah, Good catch. I missed that. Typo `Name` is using the getter whereas `name` is getting the `UnityEngine.Object` property  
Change the property name to `WeaponType` in 383882ddab43310658f66b140397e03664729be1